### PR TITLE
Documentation: add a paragraph about createAllAndroidVariantServices

### PR DESCRIPTION
**Pull request by BoD**

### CODE REVIEW

File: docs/source/advanced/plugin-configuration.mdx [hunk 1]
```diff
@@ -68,6 +68,65 @@ apollo {
...
+- main
+- demo
+    - graphql
+        - schema.graphqls // Schema for the 'demo' flavor
+        - operations.graphql // Operations specific to the 'demo' flavor
+- full
+    - graphql
+        - schema.graphqls // Schema for the 'full' flavor
+        - operations.graphql // Operations specific to the 'full' flavor
+```
```
Comment (martinbonnin) [inline docs/source/advanced/plugin-configuration.mdx:113]: very clear 👍

File: docs/source/advanced/plugin-configuration.mdx [hunk 2]
```diff
@@ -68,6 +68,65 @@ apollo {
...
+```kotlin
+apollo {
+  createAllAndroidVariantServices(".", "") {
+    if (name.startsWith("debug")) {
+      srcDir(file("src/debug/graphql/"))
+    } else if (name.startsWith("release")) {
+      srcDir(file("src/release/graphql/"))
+    }
+  }
+}
```
Comment (martinbonnin) [inline docs/source/advanced/plugin-configuration.mdx:126]: I'm not feeling too comfortable recommending `startsWith` approaches to users... 
Also, this executes codegen way too many times (basically codegenerating the same files for `debug`, `debugUnitTest` and `debugAndroidTest`)

I'd suggest something like this instead:

```kotlin
apollo {
  service("debug") {
    outputDirConnection {
      connectToAndroidSourceSet("debug")
      srcDir(file("src/debug/graphql/"))
      packageName.set("com.example")
    }
  }
  service("release") {
    outputDirConnection {
      connectToAndroidSourceSet("debug")
      srcDir(file("src/release/graphql/"))
      packageName.set("com.example")
    }
  }
}
```

I'm not 100% sure what the use case is but if I had to do it, this is how I'd do it.

Comment (BoD) [reply docs/source/advanced/plugin-configuration.mdx:126]: > codegenerating the same files for debug, debugUnitTest and debugAndroidTest

Woops you're right, not cool.

Comment (BoD) [reply docs/source/advanced/plugin-configuration.mdx:133]: I simplified the message a bit

**Review by martinbonnin — APPROVED** [review]

**Pull request merged**